src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,84 @@
+package duke;
+
+import duke.task.*;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+public class Storage {
+    private String filepath;
+    private File file;
+    
+    public Storage(String filepath) {
+        this.filepath = filepath;
+        try {
+            File file = new File(filepath);
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            this.file = file;
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+    
+    public void loadTaskListData(TaskList taskList) {
```

Comment:
You might want to break it down into smaller functions here? This function seems to be too deeply nested.